src/runtime_src/hip/api/hip_context.cpp

```diff
@@ -0,0 +1,208 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright (C) 2023-2024 Advanced Micro Device, Inc. All rights reserved.
+
+#include "hip/config.h"
+#include "hip/hip_runtime_api.h"
+
+#include "hip/core/common.h"
+#include "hip/core/context.h"
+#include "hip/core/device.h"
+
+namespace xrt::core::hip {
+// Returns handle to context
+// Throws on error
+static void
+hipCtxCreate(hipCtx_t* ctx, unsigned int flags, hipDevice_t device)
+{
+  auto hip_dev = device_cache.get(static_cast<device_handle>(device));
+  if (hip_dev == nullptr) {
```

Comment:
Change all occurrences please, I started in some other file, but too many.  
```suggestion
  if (hip_dev) {
```
As a general comment, my preference would be to use lower_case function names and more C++'ish for the functions we define (not the hip APIs of course).  In C++  this function really should return what it creates rather than use an out-variable.   It any way throws on error, which is good.

Reply:
Hi @stsoe, I have changed internal static functions implementation as suggested. Please review